File: app/src/main/java/com/albertford/autoflip/room/PageUri.java
package com.albertford.autoflip.room;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;

@Entity(foreignKeys = @ForeignKey(
    entity = Sheet.class,
    parentColumns = "id",
    childColumns = "sheetId"))
public class PageUri {
    @PrimaryKey
    private String uri;

    private int sheetId;

    // For pdfs, this is always -1
    private int pageNumber;

    public String getUri() {
        return uri;
    }

    public void setUri(String uri) {
        this.uri = uri;
    }

    public int getSheetId() {
        return sheetId;
    }

    public void setSheetId(int sheetId) {
        this.sheetId = sheetId;
    }

    public int getPageNumber() {
        return pageNumber;
    }

    public void setPageNumber(int pageNumber) {
        this.pageNumber = pageNumber;
    }
}

File: app/src/main/java/com/albertford/autoflip/room/PageUri.kt
package com.albertford.autoflip.room

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.PrimaryKey

@Entity(foreignKeys = [(ForeignKey(entity = Sheet::class, parentColumns = ["id"],
        childColumns = ["sheetId"], onDelete = ForeignKey.CASCADE))])
class PageUri (
        @PrimaryKey
        var uri: String,
        @ColumnInfo(index = true)
        var sheetId: Long,
        var pageNumber: Int
)
